### PR TITLE
Fixed multiple number support and validate after authentication

### PR DESCRIPTION
**Pull request by Kanapriya**

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java [hunk 1]
```diff
@@ -165,54 +129,176 @@ private String getLocalAuthenticatedUser(AuthenticationContext context) {
...
+    }
+
+    /**
+     * Get DUO custom error page to handle exception
+     *
+     * @param context authentication
+     * @return redirect url
+     */
+    private String getErrorPage(AuthenticationContext context) {
+        String loginPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL();
```
Comment (malakasilva) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java]: Please verify with IS team if this is the best way to do this.

Comment (malakasilva) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java]: If we deploy this webapp seperately this won't work I guess?

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java [hunk 2]
```diff
@@ -165,54 +129,176 @@ private String getLocalAuthenticatedUser(AuthenticationContext context) {
...
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("The mobile claim value and registered DUO mobile number should be in same format");
+                }
+                context.setProperty(DuoAuthenticatorConstants.NUMBER_MISMATCH, true);
+                throw new AuthenticationFailedException("Authentication failed due to mismatch in mobile numbers");
+            }
+        } else {
+            context.setProperty(DuoAuthenticatorConstants.MOBILE_CLAIM_NOT_FOUND, true);
+            throw new InvalidCredentialsException("Error while getting the mobile number from user's profile " +
```
Comment (malakasilva) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java]: Is this the proper exception in this case?

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java [hunk 3]
```diff
@@ -165,54 +129,176 @@ private String getLocalAuthenticatedUser(AuthenticationContext context) {
...
+            if (log.isDebugEnabled()) {
+                log.debug("Couldn't get the phone number of DUO user");
+            }
+            context.setProperty(DuoAuthenticatorConstants.MOBILE_NUMBER_NOT_FOUND, true);
+            throw new InvalidCredentialsException("User doesn't have a mobile number in DUO for Authentication ");
         } else {
-            return ((JSONObject) phoneArray.get(0))
-                    .getString(DuoAuthenticatorConstants.DUO_NUMBER);
+            for (int i = 0; i < phoneArray.length(); i++) {
+                if (((JSONObject) phoneArray.get(i)).getString(DuoAuthenticatorConstants.DUO_NUMBER).equals(mobile)) {
```
Comment (malakasilva) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java:217]: This can throw nullpointexception.

**Review by malakasilva — CHANGES_REQUESTED** [review]

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java (hunk 2 above)
Comment (Kanapriya) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java]: Fixed in dac301bab07386ebd20752700a6e4862afd985b1

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java (hunk 1 above)
Comment (Kanapriya) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java]: Fixed in dac301bab07386ebd20752700a6e4862afd985b1

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java (hunk 3 above)
Comment (Kanapriya) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java:217]: Before calling this method , we have check whether the mobile number is empty or not 
https://github.com/wso2-extensions/identity-outbound-auth-duo/pull/15/files#diff-318fa0251bfd40a605a65f7b90ee41c5R176

**Review by malakasilva — APPROVED** [review]

**Pull request merged**